### PR TITLE
adds more expected naming conventions to the list

### PR DESCRIPTION
**Pull request by ityonemo**
let's official-ize the following conventions:

- get, fetch, fetch!

- compare/2

### CODE REVIEW

File: lib/elixir/pages/references/naming-conventions.md [hunk 1]
```diff
@@ -124,3 +124,22 @@ When you see `length`, the operation runs in linear time ("O(n) time") because t
...
+
+### get, fetch, fetch!
+
+When you see the functions `get`, `fetch`, and `fetch!` for collections, you can expect the following behaviours:
+
+- `get` returns `nil` if the item is not present in the collection, or the requested item.
+- `fetch` returns `:error` if the item is not present in the collection, `{:ok, item}` if it is.
+- `fetch!` *raises* if the item is not present in the collection, or the requested item.
+
+Examples: `Map.get/2`, `Map.fetch/2`, `Map.fetch!/2`, `Keyword.get/2`, `Keyword.fetch/2`, `Keyword.fetch!/2`
```
Comment (josevalim) [inline lib/elixir/pages/references/naming-conventions.md:136]: I think initially we should say this is about key-value. For example, lists or enums do not have get! WDYT?

```suggestion
### get, fetch, fetch!

When you see the functions `get`, `fetch`, and `fetch!` for key-value data structures, you can expect the following behaviours:

- `get` returns `nil` if the key is not present, or the requested value.
- `fetch` returns `:error` if the key is not present, `{:ok, value}` if it is.
- `fetch!` *raises* if the key is not present, or the requested value.

Examples: `Map.get/2`, `Map.fetch/2`, `Map.fetch!/2`, `Keyword.get/2`, `Keyword.fetch/2`, `Keyword.fetch!/2`
```

File: lib/elixir/pages/references/naming-conventions.md [hunk 2]
```diff
@@ -124,3 +124,22 @@ When you see `length`, the operation runs in linear time ("O(n) time") because t
 Examples: `length/1`, `String.length/1`
 
 In other words, functions using the word "size" in its name will take the same amount of time whether the data structure is tiny or huge. Conversely, functions having "length" in its name will take more time as the data structure grows in size.
+
+### get, fetch, fetch!
+
+When you see the functions `get`, `fetch`, and `fetch!` for collections, you can expect the following behaviours:
+
+- `get` returns `nil` if the item is not present in the collection, or the requested item.
```
Comment (sabiwara) [inline lib/elixir/pages/references/naming-conventions.md:132]: Technically it returns a default, which by default is nil.

File: lib/elixir/pages/references/naming-conventions.md (hunk 1 above)
Comment (ityonemo) [reply lib/elixir/pages/references/naming-conventions.md:136]: Lovely!

File: lib/elixir/pages/references/naming-conventions.md (hunk 2 above)
Comment (ityonemo) [reply lib/elixir/pages/references/naming-conventions.md:132]: Good point!

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**